### PR TITLE
fix(cli): fix --cwd

### LINKED ISSUE #1056: [Bug] yarn breaks with relative `--cwd` option
- [ ] I'd be willing to implement a fix

**Describe the bug**

Certain yarn 2.x subcommands seem to break when passing the --cwd option.

**To Reproduce**

Given the following hierarchy: 

```
yarntest/
┣ .yarn/
┃ ┗ releases/
┃   ┗ yarn-berry.js
┣ subdir/
┃ ┗ workspace-b/
┃   ┗ package.json
┣ workspace-a/
┃ ┗ package.json
┣ .pnp.js
┣ .yarnrc.yml
┣ package.json
┗ yarn.lock
```

Attempting to pack each workspace from the the project root returns an error: 

```
$ yarn --cwd workspace-a pack
Internal Error: ENOENT: no such file or directory, lstat '/home/user/yarntest/workspace-a/workspace-a'
    at Object.realpathSync (fs.js:1529:7)
    at /home/user/yarntest/.yarn/releases/yarn-berry.js:24:103298
    at async n (/home/user/yarntest/.yarn/releases/yarn-berry.js:24:102905)
```

```
$ yarn --cwd subdir/workspace-b pack
Internal Error: ENOENT: no such file or directory, lstat '/home/user/yarntest/subdir/workspace-b/subdir'
    at Object.realpathSync (fs.js:1529:7)
    at /home/user/yarntest/.yarn/releases/yarn-berry.js:24:103298
    at async n (/home/user/yarntest/.yarn/releases/yarn-berry.js:24:102905)
```

**Environment if relevant (please complete the following information):**

 - OS: Linux
 - Node version: 12.15.0
 - Yarn version: 2.0.0-rc.29

**Additional context**

Passing an absolute path succeeds, but only when the subdirectory is a workspace folder (which is 
a separate issue #705).

### PR DESCRIPTION
**Pull request by paul-soporan**
**What's the problem this PR addresses?**

There are 2 code paths here: https://github.com/yarnpkg/berry/blob/215455f2c303a14fcb10e5513bdbbf82b5320d3d/packages/yarnpkg-cli/sources/main.ts#L64
The `--cwd` option is only checked for on the `else` branch, so when [`run`](https://github.com/yarnpkg/berry/blob/215455f2c303a14fcb10e5513bdbbf82b5320d3d/packages/yarnpkg-cli/sources/main.ts#L94) is called after `--cwd` is encountered, there is a possibility that the `if` branch will execute with the `--cwd` option that was already encountered.

This causes commands to be executed in the wrong directory in some cases.

This PR closes #1056.

**How did you fix it?**

I made the `--cwd` flag and its argument get removed from `process.argv` when `--cwd` is encountered.

### CODE REVIEW

File: packages/yarnpkg-cli/sources/main.ts [hunk 1]
```diff
@@ -90,6 +90,7 @@ export async function main({binaryVersion, pluginConfiguration}: {binaryVersion:
         const iShouldBeHere = realpathSync(cwd);
 
         if (iAmHere !== iShouldBeHere) {
+          process.argv.splice(process.argv.findIndex((arg) => arg === `--cwd`), 2);
```
Comment (arcanis) [inline packages/yarnpkg-cli/sources/main.ts:93]: Not a fan of the slice. In particular, someone running `--cwd=foo` will hit weird problems 🤔

Can we just add a parameter to run (`enableCwd`, defaults to true) that we would set to `false` in the recursive call?

Comment (paul-soporan) [reply packages/yarnpkg-cli/sources/main.ts:93]: Totally forgot that `--cwd=foo` is also valid.
There's a problem with the `enableCwd` approach: It still only gets checked on the else branch, so the issue wouldn't be solved. I don't see a way for it to be passed to the if branch without changing a lot of things.

Comment (arcanis) [reply packages/yarnpkg-cli/sources/main.ts:93]: And an environment variable, similar to what we do with the `YARN_PATH`? For example we could set `YARN_IGNORE_CWD`, and delete it in subprocesses.

A repro test would also help to better understand under which circumstances the bug is triggered

Comment (paul-soporan) [reply packages/yarnpkg-cli/sources/main.ts:93]: The bug is triggered every time a relative `--cwd` (the person who opened #1134 says that absolute paths work, I haven't tested it myself) is provided:
```js repro
const {promises: {mkdir}} = require(`fs`);

await mkdir(`./subdir`);

await yarn(`./subdir`, `init`);

await expect(yarn(`--cwd`, `./subdir`)).rejects.not.toThrow(`ENOENT: no such file or directory, lstat`);
```
I'll try the environment variable approach.

Comment (paul-soporan) [reply packages/yarnpkg-cli/sources/main.ts]: The environment variable approach works.😀

**Comment by arcanis** [comment]
🚀

**Pull request merged**